src/main/java/uk/ac/ebi/ena/readtools/fastq/ena/Fastq2Sam.java

```diff
@@ -0,0 +1,204 @@
+import uk.ac.ebi.ena.readtools.common.reads.normalizers.htsjdk.IlluminaQualityNormalizer;
+import uk.ac.ebi.ena.readtools.common.reads.normalizers.htsjdk.SolexaQualityNormalizer;
+import uk.ac.ebi.ena.readtools.common.reads.normalizers.htsjdk.StandardQualityNormalizer;
+import uk.ac.ebi.ena.readtools.loader.common.FileCompression;
+import uk.ac.ebi.ena.readtools.loader.common.consumer.DataConsumer;
+import uk.ac.ebi.ena.readtools.loader.common.producer.AbstractDataProducer;
+import uk.ac.ebi.ena.readtools.loader.common.producer.DataProducerException;
+import uk.ac.ebi.ena.readtools.loader.fastq.DataSpot;
+import uk.ac.ebi.ena.readtools.loader.fastq.DataSpot.DataSpotParams;
+import uk.ac.ebi.ena.readtools.loader.fastq.PairedFastqConsumer;
+import uk.ac.ebi.ena.readtools.loader.fastq.SingleFastqConsumer;
+import uk.ac.ebi.ena.readtools.loader.fastq.IlluminaSpot;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Fastq2Sam {
+
+    public static void main( String[] args ) {
+        final Params p = new Params();
+        JCommander jc = new JCommander( p );
+        try {
+            jc.parse( args );
+        } catch( ParameterException pe ) {
+            jc.usage();
+            System.exit( Params.PARAM_ERROR );
+        }
+        
+        if( p.help ) {
+            jc.usage();
+            System.exit( Params.OK_CODE );
+        }
+        
+        try {
+            new Fastq2Sam().create( p );
+            System.exit( Params.OK_CODE );
+            
+        } catch( Throwable e ) {
+            e.printStackTrace();
+            System.exit( Params.FAILURE );
+        }
+    }
+    
+    public void create( Params p ) throws NoSuchMethodException, IOException {
+        DataConsumer<DataSpot, IlluminaSpot> spotToIlluminaSpotConsumer = null;
+
+        if( null == p.files || p.files.size() < 1 || p.files.size() > 2) {
+            throw new IllegalArgumentException("Invalid number of input files : " + p.files.size());
+        } else if( 1 == p.files.size() ) {
+            //single
+            spotToIlluminaSpotConsumer = (DataConsumer<DataSpot, IlluminaSpot>)new SingleFastqConsumer();
+        } else if( 2 == p.files.size() ) {
+            //same file names;
+            if( p.files.get( 0 ).equals( p.files.get( 1 ) ) ) {
+                throw new IllegalArgumentException(
+                        "Paired files cannot be same. File1 : " + p.files.get(0) + ", File2 : " + p.files.get(1));
+            }
+
+            spotToIlluminaSpotConsumer = (DataConsumer<DataSpot, IlluminaSpot>) new PairedFastqConsumer(
+                    new File( p.tmp_root ), p.spill_page_size );
+        }
+        
+        if( p.verbose ) {
+            System.out.println( "Files to process: " );
+            for( String f_name: p.files )
+                System.out.println( " " + f_name );
+        }
+
+        Fastq2BamConsumer illuminaSpotToBamConsumer = new Fastq2BamConsumer(

```

Comment:
Variable name should be 'fastq...' instead of 'illumina...'. Applies to all similar cases.